error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ClassRules.java

```diff
@@ -62,4 +63,31 @@ Predicate<S> after(Class<T> clazz) {
       return clazz::isInstance;
     }
   }
+
+  /** Prefer {@link Class#cast(Object)} method references over more verbose alternatives. */
```

Comment:
"Verbose" is perhaps a bit strong/unclear. I see that this mirrors the rules above, though. Will propose something.